### PR TITLE
fix: properly handle template literal types

### PR DESCRIPTION
**Pull request by kitsonk**
Fixes: #135

This changes the format of the `"kind": "literal"` serialization of `tsType`, as the structure was not able to properly represent template literal types, so downstream consumers of the `DocNode` structure or serialization.

The `DocPrinter` though prints out the changes "properly".

### CODE REVIEW

File: src/ts_type.rs [hunk 1]
```diff
@@ -1,52 +1,27 @@
...
-  TsThisTypeOrIdent, TsTupleType, TsType, TsTypeAnn, TsTypeLit, TsTypeOperator,
-  TsTypeParamInstantiation, TsTypePredicate, TsTypeQuery, TsTypeRef,
-  TsUnionOrIntersectionType, VarDeclarator,
-};
-
-// pub enum TsType {
-//  *      TsKeywordType(TsKeywordType),
-//  *      TsThisType(TsThisType),
-//  *      TsFnOrConstructorType(TsFnOrConstructorType),
-//  *      TsTypeRef(TsTypeRef),
```
Comment (bartlomieju) [inline src/ts_type.rs:25]: Thanks for cleaning this up

File: lib/types.d.ts [hunk 2]
```diff
@@ -295,14 +301,19 @@ export interface LiteralDefNumber extends LiteralDefBase {
 
 export interface LiteralDefBigInt extends LiteralDefBase {
   kind: "bigInt";
-  number: bigint;
+  string: string;
```
Comment (bartlomieju) [inline lib/types.d.ts:304]: Is this change on purpose?

**Review by bartlomieju — APPROVED** [review]
LGTM, it's a bit unfortunate that it's a breaking change

File: lib/types.d.ts (hunk 2 above)
Comment (kitsonk) [reply lib/types.d.ts:304]: Yeah, it was a mistake on my part, and I didn't notice it until I was making the other changes.  Silly me forgot there is no way to serialize a bigint to JSON as a bigint.

**Comment by kitsonk** [comment]
It is a breaking change, because the original implementation was sadly incorrect on several levels.
